crates/bevy_ecs/src/schedule_v3/executor/multi_threaded.rs

```diff
@@ -0,0 +1,506 @@
+use bevy_tasks::{ComputeTaskPool, Scope, TaskPool};
+use bevy_utils::default;
+use bevy_utils::syncunsafecell::SyncUnsafeCell;
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::{info_span, Instrument};
+
+use async_channel::{Receiver, Sender};
+use fixedbitset::FixedBitSet;
+
+use crate::{
+    archetype::ArchetypeComponentId,
+    query::Access,
+    schedule_v3::{
+        is_apply_system_buffers, BoxedCondition, ExecutorKind, SystemExecutor, SystemSchedule,
+    },
+    world::World,
+};
+
+/// Per-system data used by the [`MultiThreadedExecutor`].
+// Copied here because it can't be read from the system when it's running.
+struct SystemTaskMeta {
```

Comment:
Nit: this would be much easier to parse if it was `SystemTaskMetadata`, at least for me.